sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -565,16 +566,31 @@ class depends_on {
   public:
     template<typename... NodeTN>
     depends_on(NodeTN... nodes);
+
+    template<typename... EventTN>
+    depends_on(EventTN... events);
 };
 }
 ----
 
 The API for explicitly adding nodes to a `command_graph` includes a
 `property_list` parameter. This extension defines the `depends_on` property to
```

Comment:
Creating a new thread to branch out of the https://github.com/reble/llvm/pull/372/files#r1647663947 thread, rather than taking over that with a tangent.

> Considering this more, I'm concerned that get_event(node) may not be implementable on a CUDA backend. Since the executable graph has already been finalized then we can't add nodes to it, but to implement this ontop of CUDA-Graph I think we need to use cudaGraphAddEventRecordNode() - which would require some mechanism in the modifiable graph to say that a node is going to produce a sycl event.

I think we need a mechanism for the user to tell the runtime that a node in a modifiable graph will signal out of the graph, so that when we finalize the graph we can create a `cudaGraphAddEventRecordNode` to do that.

My first thought is to add a property that can be set on a node to enable this, and then only nodes with this property can be used in `get_event()`. My other idea was a new node type, but that is effectively an empty node with this property set. Open to other ideas, or discussion on whether the current design is indeed unsuited to a CUDA backend.

Reply:
Actually for L0 we could still use this info for more precise `zeCommandListGetNextCommandIdExp` usage so that we are only setting the `ZE_MUTABLE_COMMAND_EXP_FLAG_SIGNAL_EVENT` and `ZE_MUTABLE_COMMAND_EXP_FLAG_WAIT_EVENTS` on commands which the users has stated will be used with dynamic events.

Reply:
After some internal discussion on this topic we settled on the idea of:

- A node property for the explicit API which marks a node as requiring a backend event which can be obtained via `get_event()`
- A graph property usable with `command_graph::begin_recording()` which opts-in to this for all recorded nodes from a queue/queues until `end_recording()` is called.

This allows fine grained opt-in to these required events on both APIs.

Reply:
This sounds reasonable, but I wouldn't call this a "backend event".  It's just an `event`.  Calling it "backend event" makes it sound like it returns a native event from the backend, which is not the case.

Reply:
Good point! I will avoid such phrasing in the wording here.